src/main/scala/com/databricks/spark/redshift/Parameters.scala

```diff
@@ -0,0 +1,183 @@
+     */
+    def distStyle = parameters.get("diststyle")
+
+    /**
+     * The name of a column in the table to use as the distribution key when using DISTSTYLE KEY.
+     * Not set by default, as default DISTSTYLE is EVEN.
+     */
+    def distKey = parameters.get("distkey")
+
+    /**
+     * A full Redshift SORTKEY specification. For full information, see latest Redshift docs:
+     * http://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_TABLE_NEW.html
+     *
+     * Examples:
+     *   SORTKEY (my_sort_column)
+     *   COMPOUND SORTKEY (sort_col1, sort_col2)
+     *   INTERLEAVED SORTKEY (sort_col1, sort_col2)
+     *
+     * Not set by default - table will be unsorted.
+     *
+     * Note: appending data to a table with a sort key only makes sense if you know that the data being added
+     * will be after the data already in the table according to the sort order. Redshift does not support random
+     * inserts according to sort order, so performance will degrade if you try this.
+     */
+    def sortKeySpec = parameters.get("sortkeyspec")
+
+    /**
+     * When true, data is always loaded into a new temporary table when performing an overwrite.
+     * This is to ensure that the whole load process succeeds before dropping any data from Redshift,
+     * which can be useful if, in the event of failures, stale data is better than no data for your systems.
+     *
+     * Defaults to true.
+     */
+    def useStagingTable = parameters("usestagingtable").toBoolean
+
+    /**
+     * List of semi-colon separated SQL statements to run after successful write operations.
+     * This can be useful for running GRANT operations to make your new tables readable to other users and groups.
+     *
+     * If the action string contains %s, the table name will be substituted in, in case a staging table is being used.
+     *
+     * Defaults to empty.
+     */
+    def postActions = parameters("postactions").split(";")
+
+    /**
+     * Looks up "aws_access_key_id" and "aws_secret_access_key" in the parameter map
+     * and generates a credentials string for Redshift. If no credentials have been provided,
+     * this function will instead try using the AWS DefaultCredentialsProviderChain, which makes
+     * use of standard system properties, environment variables, or IAM role configuration if available.
+     */
+    def credentialsString() = {
+
+      val (accessKeyId, secretAccessKey) =
+        if(parameters.contains("aws_access_key_id")) {
+          (parameters("aws_access_key_id"), parameters("aws_secret_access_key"))
+        } else {
+          try {
+            val awsCredentials = (new DefaultAWSCredentialsProviderChain).getCredentials
+            (awsCredentials.getAWSAccessKeyId, awsCredentials.getAWSSecretKey)
+          } catch {
+            case e: Exception => throw new Exception("No credentials provided and unable to detect automatically.", e)
```

Comment:
Do you mean something like `s3://user:pass@bucket/.../`? I'm pretty sure Redshift doesn't allow that in `COPY` or `UNLOAD`, as the syntax for those commands says `CREDENTIALS` is a mandatory parameter: http://docs.aws.amazon.com/redshift/latest/dg/r_COPY.html#r_COPY-syntax-overview-credentials

I guess we could potentially parse the URI, extract the credentials and then reformat it without them or something? Getting complicated though!

I've added support for reading `fs.<scheme>.*` from the hadoopConfiguration, where scheme is determined by `tempdir`.